cpp/log/logged_entry.h

```diff
@@ -53,12 +53,18 @@ class LoggedCertificate : public ct::LoggedCertificatePB {
   }
 
   bool SerializeExtraData(std::string* dst) const {
-    if (entry().type() == ct::X509_ENTRY)
-      return Serializer::SerializeX509Chain(entry().x509_entry(), dst) ==
-             Serializer::OK;
-    else
-      return Serializer::SerializePrecertChainEntry(entry().precert_entry(),
-                                                    dst) == Serializer::OK;
+    switch (entry().type()) {
+      case ct::X509_ENTRY:
+        return Serializer::SerializeX509Chain(entry().x509_entry(), dst) ==
+               Serializer::OK;
+        break;
+      case ct::PRECERT_ENTRY:
+        return Serializer::SerializePrecertChainEntry(entry().precert_entry(),
+                                                      dst) == Serializer::OK;
+        break;
+      default:
+        LOG(FATAL) << "Unknown entry type: " << entry().type();
+    }
```

Comment:
Could you use the pattern where each case calls `return`, and not use a `default` clause, so that compiler give us warnings on unhandled `enum` values?